include/kodlab_mjbots_sdk/attitude.h

```diff
@@ -0,0 +1,363 @@
+      att_rot_mat_ = {Eigen::Matrix3<Scalar>::Identity(), false};;
+
+  /**
+   * @brief Angular velocity about <code>[x, y, z]</code> axes (default deg/s)
+   */
+  Eigen::Vector3<Scalar> ang_rate_ = {0, 0, 0};
+
+  /**
+   * @brief Linear acceleration along <code>[x, y, z]</code> axes (default
+   *        m/s^2)
+   */
+  Eigen::Vector3<Scalar> accel_ = {0, 0, 0};
+
+  /**
+   * @brief Gyroscope bias about <code>[x, y, z]</code> axes (default deg/s)
+   */
+  Eigen::Vector3<Scalar> ang_rate_bias_ = {0, 0, 0};
+
+  /**
+   * @brief Attitude uncertainty about the raw attitude
+   */
+  Eigen::Quaternion<Scalar> att_uncertainty_ = {1, 0, 0, 0};
+
+  /**
+   * @brief Gyroscope bias uncertainty about <code>[x, y, z]</code> axes
+   * (default deg/s)
+   */
+  Eigen::Vector3<Scalar> ang_bias_uncertainty_ = {0, 0, 0};
+
+  /**
+   * @brief World frame rotation offset
+   * @note Useful to change the direction of gravity
+   */
+  Eigen::Quaternion<Scalar> world_offset_ = {1, 0, 0, 0};
+
+  /**
+   * @brief Invalidates cached data
+   */
+  void InvalidateCached()
+  {
+    att_euler_.invalidate();
+    att_rot_mat_.invalidate();
+  }
+
+public:
+  /**
+   * @brief Construct a new \c Attitude object.
+   */
+  Attitude() = default;
+
+  /**
+   * @brief Construct a new \c Attitude object with populated data.
+   *
+   * @param att_quat_in attitude quaternion
+   * @param ang_rate_in angular velocity about <code>[x, y, z]</code> axes
+   * @param accel_in linear acceleration along <code>[x, y, z]</code> axes
+   * @param world_offset_in world frame rotation offset (default identity)
+   */
+  Attitude(const Eigen::Quaternion<Scalar> &att_quat_in,
+           const Eigen::Vector3<Scalar> &ang_rate_in,
+           const Eigen::Vector3<Scalar> &accel_in,
+           const Eigen::Quaternion<Scalar> &world_offset_in = {1, 0, 0, 0})
+      : att_quat_raw_(att_quat_in),
+        att_quat_(world_offset_in * att_quat_in),
+        ang_rate_(ang_rate_in),
+        accel_(accel_in),
+        world_offset_(world_offset_in) {}
+
+  /**
+   * @brief Construct a new \c Attitude object with populated data, biases,
+   *        and uncertainty.
+   *
+   * @param att_quat_in attitude quaternion
+   * @param ang_rate_in angular velocity about <code>[x, y, z]</code> axes
+   * @param accel_in linear acceleration along <code>[x, y, z]</code> axes
+   * @param ang_rate_bias_in gyroscope bias about <code>[x, y, z]</code> axes
+   * @param att_uncertainty_in attitude uncertainty quaternion
+   * @param ang_bias_uncertainty_in gyroscope bias uncertainty about
+   *                                <code>[x, y, z]</code> axes
+   * @param world_offset_in world frame rotation offset (default identity)
+   */
+  Attitude(const Eigen::Quaternion<Scalar> &att_quat_in,
+           const Eigen::Vector3<Scalar> &ang_rate_in,
+           const Eigen::Vector3<Scalar> &accel_in,
+           const Eigen::Vector3<Scalar> &ang_rate_bias_in,
```

Comment:
This is so that the `IMUData` class world offset can be easily modified both in cases where only {attitude, ang. velocity, acceleration} are used, as well as those usages including uncertainty and biases.

Reply:
This is another case pushing for having a simple attitude struct inside or along side of imu_data so as to separate those parts of the class intuitively. [Prev](https://github.com/KodlabPenn/kodlab_mjbots_sdk/pull/16/files/61bd4eed34b1371f16943b3f609522a114a45c50#r935094001)

Reply:
@jdcaporale make an issue out of this.